### PR TITLE
Fixed text around entity types

### PR DESCRIPTION
**Pull request by aaronjeline**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/schema.md [hunk 1]
```diff
@@ -117,19 +117,27 @@ If the parent type is part of the same namespace as the child type, then you can
 
 Specifies the data type and attributes that are needed to define entities of this type. 
 
-Each attribute defined in a `shape` must include a `type` specification, with a value that names one of the [Cedar supported data types](syntax-datatypes.md). The following example shows a simple shape.
+The top level `shape` of an entity must have type `Record`. The follwoing example shows a simple Entity specification with a shape:
```
Comment (mwhicks1) [inline docs/schema.md:121]: Change to "The following [notice typo fix] example shows a simple specification of the `User` entity type:"

File: docs/schema.md [hunk 2]
```diff
@@ -117,19 +117,27 @@ If the parent type is part of the same namespace as the child type, then you can
...
 
-By default, attributes that you define are required. You can make an attribute optional by adding `"required": false` to the attribute. The following example shows an attribute called `jobLevel` that is an optional attribute for whatever entity it's part of. You can also explicitly declare that an attribute is mandatory by including `"required": true` to the shape.
+Each attribute in a `Record` can be required or optional. If required, then policies that reference this type can assume that the attribute is always present. If optional, then a policy should check or the attribute's presence by using the [has](syntax-operators.md#operator-has) operator before trying to access the attribute's value. If evaluation of a policy results in an attempt to access a non-existent attribute, Cedar generates an exception. You can make an attribute optional by adding `"required": false` to the attribute. The following example shows an attribute called `jobLevel` that is an optional attribute for whatever entity it's part of. You can also explicitly declare that an attribute is mandatory by including `"required": true` to the shape.
```
Comment (mwhicks1) [inline docs/schema.md:140]: Please restore "By default, attributes that you define are required." at the start. It's less clear in the text you wrote. The rest looks great!

**Review by mwhicks1 — CHANGES_REQUESTED** [review]

**Review by anwarmamat — APPROVED** [review]

**Comment by mwhicks1** [comment]
Also: The text that follows the indication of `shape` in the docs could use revision. It still reads as if something other than `Record` is a reasonable choice. E.g., around line 149 it says "use type `Record` when ..." but of course Record is the only actual choice. Probably you will need to make a bunch of changes here, perhaps even deleting big chunks of text.

File: docs/schema.md [hunk 3]
```diff
@@ -138,12 +146,12 @@ By default, attributes that you define are required. You can make an attribute o
 },
 ```
 
-If the entity type for an attribute in the shape needs to be a more complex structure instead of a simple value, then you can define the element as being of one of the types [Record](#schema-entitytypes-shape-record), [Set](#schema-entitytypes-shape-set), or [Entity](#schema-entitytypes-shape-entity).
+Some attributes types require further description as to the values they contain:
```
Comment (aaronjeline) [inline docs/schema.md:149]: Do we need to explicitly mention types can be arbitrarily nested?

File: docs/schema.md [hunk 4]
```diff
@@ -117,19 +117,27 @@ If the parent type is part of the same namespace as the child type, then you can
...
+            "age" : {
+                "type" : "Long"
+            }
+        }
     }
 }
 ```
 
-Each attribute in a `shape` can be required or optional. If required, then policies that reference this type can assume that the attribute is always present. If optional, then a policy should check or the attribute's presence by using the [has](syntax-operators.md#operator-has) operator before trying to access the attribute's value. If evaluation of a policy results in an attempt to access a non-existent attribute, Cedar generates an exception.
+Each attribute defined in a `shape` must include a `type` specification, with a value that names one of the [Cedar supported data types](syntax-datatypes.md).
```
Comment (mwhicks1) [inline docs/schema.md:138]: Each attribute defined in a `shape`'s record

**Review by mwhicks1 — CHANGES_REQUESTED** [review]

**Pull request merged**